crates/miden-objects/src/transaction/executed_tx.rs

```diff
@@ -202,35 +183,43 @@ impl From<ExecutedTransaction> for TransactionMeasurements {
 
 impl Serializable for ExecutedTransaction {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
-        self.tx_inputs.write_into(target);
+        self.tx_inputs.account().write_into(target);
+        self.tx_inputs.block_header().write_into(target);
+        self.tx_inputs.blockchain().write_into(target);
+        self.tx_inputs.input_notes().write_into(target);
+        self.tx_inputs.tx_args().write_into(target);
+        self.tx_inputs.advice_inputs().write_into(target);
+        self.tx_inputs.foreign_account_inputs().write_into(target);
```

Comment:
It may be better to implement `Serializable`/`Deserializable` for `TransactionInputs` rather than do it manually here. Unless there is a good reason not to do it?